plugins/jdbc-adapter-framework/src/main/java/org/polypheny/db/adapter/jdbc/stores/AbstractJdbcStore.java

```diff
@@ -119,254 +138,270 @@ public void createUdfs() {
+        if ( column.length != null && doesTypeUseLength( column.type ) ) {
+            builder.append( "(" );
+            builder.append( column.length );
+            if ( column.scale != null ) {
+                builder.append( "," ).append( column.scale );
             }
-            executeUpdate( builder, context );
+            builder.append( ")" );
         }
+        executeUpdate( builder, context );
+
+        updateNativePhysical( allocId );
+
+    }
+
+
+    public boolean doesTypeUseLength( PolyType type ) {
+        return type != PolyType.TEXT;
     }
 
 
     @Override
-    public void dropTable( Context context, CatalogTable catalogTable, List<Long> partitionIds ) {
+    public void dropTable( Context context, long allocId ) {
         // We get the physical schema / table name by checking existing column placements of the same logical table placed on this store.
         // This works because there is only one physical table for each logical table on JDBC stores. The reason for choosing this
         // approach rather than using the default physical schema / table names is that this approach allows dropping linked tables.
-        String physicalTableName;
-        String physicalSchemaName;
-
-        List<CatalogPartitionPlacement> partitionPlacements = new ArrayList<>();
-        partitionIds.forEach( id -> partitionPlacements.add( catalog.getPartitionPlacement( getAdapterId(), id ) ) );
+        PhysicalTable table = adapterCatalog.fromAllocation( allocId );
+        StringBuilder builder = new StringBuilder();
 
-        for ( CatalogPartitionPlacement partitionPlacement : partitionPlacements ) {
-            catalog.deletePartitionPlacement( getAdapterId(), partitionPlacement.partitionId );
-            physicalSchemaName = partitionPlacement.physicalSchemaName;
-            physicalTableName = partitionPlacement.physicalTableName;
+        builder.append( "DROP TABLE " )
+                .append( dialect.quoteIdentifier( table.namespaceName ) )
+                .append( "." )
+                .append( dialect.quoteIdentifier( table.name ) );
 
-            StringBuilder builder = new StringBuilder();
+        if ( RuntimeConfig.DEBUG.getBoolean() ) {
+            log.info( "{} from store {}", builder, this.getUniqueName() );
+        }
+        executeUpdate( builder, context );
+        adapterCatalog.removeAllocAndPhysical( allocId );
+    }
 
-            builder.append( "DROP TABLE " )
-                    .append( dialect.quoteIdentifier( physicalSchemaName ) )
-                    .append( "." )
-                    .append( dialect.quoteIdentifier( physicalTableName ) );
 
-            if ( RuntimeConfig.DEBUG.getBoolean() ) {
-                log.info( "{} from store {}", builder.toString(), this.getUniqueName() );
-            }
-            executeUpdate( builder, context );
-        }
+    @Override
+    public void renameLogicalColumn( long id, String newColumnName ) {
```

Comment:
Why is there a method to update the logical column name in the adapter? Is this a mistake and should refer to the physical column?